### ISSUE TITLE
Can't Train or Get Inference while Batch Size = 1

### ISSUE THREAD

**Issue by formidiable** [MehranTaghian/DQN-Trading#4]
I searched a lot but couldn't find a proper way to fix this.

![2mm_5mm_composed](https://user-images.githubusercontent.com/77826094/151828880-5a24149c-5345-4da6-89cd-44c404859375.png)

I believe Batch Normalization cause the problem. Do you have any advice about it ?

**Comment by formidiable** [comment]
There is an explanation about my suspicion about encoder decoder and usage Batch Normalization in this link.  [^](https://stackoverflow.com/a/69286469)

**Comment by formidiable** [comment]
Layer Normalization technique fixed the training issue.

**Comment by formidiable** [comment]
I also got this error while training without batchnorm1d or train with layernorm.

_RuntimeError: There were no tensor arguments to this function (e.g., you passed an empty list of Tensors), but no fallback function is registered for schema aten::_cat.  This usually means that this function requires a non-empty list of Tensors.  Available functions are [CPU, CUDA, QuantizedCPU, BackendSelect, Named, AutogradOther, AutogradCPU, AutogradCUDA, AutogradXLA, AutogradPrivateUse1, AutogradPrivateUse2, AutogradPrivateUse3, Tracer, Autocast, Batched, VmapMode]._
